pkg/apis/alibabacloudprovider/v1beta1/alibabacloudmachineproviderconfig_types.go

```diff
@@ -217,17 +209,21 @@ type AlibabaCloudMachineProviderConfig struct {
 	// CredentialsSecret is a reference to the secret with alibabacloud credentials. Otherwise, defaults to permissions
 	// provided by attached RAM role where the actuator is running.
 	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
+
+	// Tags are the set of metadata to add to an instance.
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
 }
 
 // ResourceTagReference is a reference to a specific AlibabaCloud resource by ID, or tags.
 // Only one of ID or Tags may be specified. Specifying more than one will result in
 // a validation error.
-type ResourceTagReference struct {
+type AlibabaResourceReference struct {
 	// ID of resource
 	// +optional
 	ID string `json:"id,omitempty"`
 
-	// Tags is a set of tags used to identify a resource
+	// Filters is a set of metadata based upon ECS object tags used to identify a resource
 	Tags []Tag `json:"tags,omitempty"`
```

Comment:
This comment doesn't match the field name, did we intend to rename the filed?

Reply:
I was going to go filters but `Tags` are actually used and the structure for AWS didn't align here. AWS can have a single `key` and multiple `values`. Where as here, it is 1:1 for `key`:`value`. I think `Tags` is appropriate.

Reply:
Will rename.